### PR TITLE
Add dactyl_manuform_rp2040

### PR DESCRIPTION
**Pull request by GustawXYZ**
Adds new keyboard under handwired/dactyl_manuform_rp2040

It is a mashup of alcor_datyl nad dactyl_manuform(5x6)

## Description

Started by cloning alcor_dactyl. Changed GPIOs to be Raspberry Pi Pico friendly.
Changed the layout to use dactyl_manuform 5x6. Added my default keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [NA?] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl_manuform_rp2040/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#if defined(ENCODER_ENABLE) && defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+
+};
+#endif // defined(ENCODER_ENABLE) && defined(ENCODER_MAP_ENABLE)
+
+
+
+
+
```
Comment (dunk2k) [inline keyboards/handwired/dactyl_manuform_rp2040/keymaps/default/keymap.c:73]: code block not required as encoders not set to true and the code block is essentially null/functionless
```suggestion
```

File: keyboards/handwired/dactyl_manuform_rp2040/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 QMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+#define _QWERTY 0
+#define _NUMPAD 1
+#define _ARROWS 2
+#define _MOUSE 3
+#define _GAMING 4
```
Comment (dunk2k) [inline keyboards/handwired/dactyl_manuform_rp2040/keymaps/default/keymap.c:10]:
```suggestion
enum layers {
    _QWERTY,
    _NUMPAD,
    _ARROWS,
    _MOUSE,
    _GAMING,
};
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
@GustawXYZ 

Would you be able to change `dactyl_manuform_rp2040` to `dactyl_manuform_pi_pico` as not to confuse terminology with QMK's rp2040 based converters and rp2040 development boards?

File: keyboards/handwired/dactyl_manuform_rp2040/info.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    "split": {
+        "enabled": true
+    },
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": ["GP10", "GP11", "GP12", "GP13", "GP14", "GP15"],
+        "rows": ["GP21", "GP20", "GP19", "GP18", "GP17", "GP16"]
+    },
+    "rgb_matrix": {
+        "driver": "WS2812",
```
Comment (dunk2k) [inline keyboards/handwired/dactyl_manuform_rp2040/info.json:31]:
```suggestion
        "driver": "WS2812",
        "split_count": [1, 1],
```

File: keyboards/handwired/dactyl_manuform_rp2040/config.h [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+#pragma once
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET_TIMEOUT 200U
+#define SERIAL_USART_FULL_DUPLEX
+#define SERIAL_USART_TX_PIN GP9
+#define SERIAL_USART_RX_PIN GP8
+#define RGB_MATRIX_LED_COUNT 2
+#define RGB_MATRIX_SPLIT \
+    { 1, 1 }
```
Comment (dunk2k) [inline keyboards/handwired/dactyl_manuform_rp2040/config.h:12]: moved to `info.json`
```suggestion
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/handwired/dactyl_manuform_pi_pico/readme.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+Build instructions[WIP]: https://github.com/GustawXYZ/dactyl_manuform_rp2040/ 
+
+* Keyboard Maintainer: [Gustaw.xyz](https://github.com/Gustaw.xyz)
+* Hardware Supported: Raspberry Pi Pico and other RP2040
+* Hardware Availability: https://www.raspberrypi.com/products/raspberry-pi-pico/
+* 3D Print model: https://github.com/abstracthat/dactyl-manuform
+
+Make example for this keyboard (after setting up your build environment):
+
+    make dactyl_manuform_pi_pico:default
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform_pi_pico/readme.md:18]:
```suggestion
    make handwired/dactyl_manuform_pi_pico:default
```

File: keyboards/handwired/dactyl_manuform_pi_pico/readme.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Flashing example for this keyboard:
+
+    make dactyl_manuform_pi_pico:default:flash
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform_pi_pico/readme.md:22]:
```suggestion
    make handwired/dactyl_manuform_pi_pico:default:flash
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/dactyl_manuform_pi_pico/config.h [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET_TIMEOUT 200U
+#define SERIAL_USART_FULL_DUPLEX
+#define SERIAL_USART_TX_PIN GP9
+#define SERIAL_USART_RX_PIN GP8
+#define RGB_MATRIX_LED_COUNT 2
+#define EE_HANDS
```
[the remark this answers is not in the document]
Comment (GustawXYZ) [reply keyboards/handwired/dactyl_manuform_pi_pico/config.h:11]: Thanks for the pointer. I don't think I've noticed any issues using the keyboard for last month or two. Although pi pico has plenty GPIO. What would be the symtomps of this? Bacause I'm not sure if it's worth resoldering

File: keyboards/handwired/dactyl_manuform_pi_pico/info.json [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    "split": {
+        "enabled": true
+    },
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": ["GP10", "GP11", "GP12", "GP13", "GP14", "GP15"],
+        "rows": ["GP21", "GP20", "GP19", "GP18", "GP17", "GP16"]
+    },
+    "rgb_matrix": {
+        "driver": "WS2812",
```
[the remark this answers is not in the document]
Comment (GustawXYZ) [reply keyboards/handwired/dactyl_manuform_pi_pico/info.json:31]: Thanks a lot! I was going to look into that in some spare time

File: keyboards/handwired/dactyl_manuform_pi_pico/config.h (hunk 7 above)
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/handwired/dactyl_manuform_pi_pico/config.h:11]: Mostly, user experience.  If it's working fine, that that's okay.

**Pull request closed without merging**